src/meta/Matches.ts

```diff
@@ -0,0 +1,14 @@
+import { Base } from './Base';
+
+export default class Matches extends Base {
+	/**
+	 * Determine if the target of a particular `Event` matches the virtual DOM key
+	 * @param key The virtual DOM key
+	 * @param event The event object
+	 */
+	public get(key: string, event: Event): boolean {
+		this.requireNode(key);
```

Comment:
@kitsonk it's quite difficult to explain the scenario that I think will be unexpected as a user, I put together an example repo where errors are thrown because user actions cause nodes to be removed - in a way that I wouldn't think are expected... https://github.com/agubler/dojo2-matches-meta

Needs your branch of course :)